rsc/src/main/scala/rsc/semanticdb/Writer.scala

```diff
@@ -59,8 +59,25 @@ final class Writer private (
 
   private def validate(outline: Outline, info: s.SymbolInformation): Unit = {
     (outline, info.signature) match {
+
       case (defn: DefnMethod, m: s.MethodSignature) if m.returnType.isEmpty =>
         reporter.append(DefnMethodNotype(defn, settings.notypeWarn))
+
+      case (defn: DefnClass, _: s.ClassSignature) =>
```

Comment:
Also: add comment
```
// Check that the code doesn't need type parameters to parent classes/traits to be inferred.
// (Scalac can infer type parameters for the first parent of a class, while Rsc can't.
// If we didn't check for this here, Rsc would emit outlines with type parameters that
// should be inferred simply omitted.)
```

Reply:
I don't think we want to go down the slippery slope of developing extra functionality without much user feedback (even if I like the idea of catching errors earlier).  We can open a ticket and then deal with it when appropriate. This ticket was mainly solving the problem that scalac's error message was cryptic/useless, whereas for the `with T` code sample would result in a useful one:
```scala
error: trait T takes type parameters
class D extends C[Int](1) with T
```

Reply:
That's fine. I'd thought about that possibility as well.

One further suggestion: instead of
`defn.parents.headOption match { case Some(init @ Init(tpt: TptId, _)) => ... }`
how about
`defn.parents match { case (init @ Init(tpt: TptId, _)) :: _ => ... }`

Reply:
that's really nice!